### PR TITLE
Treap priority

### PR DESCRIPTION
**Pull request by dented42**
I added some functions that allow a user to manipulate node priorities after finding myself wishing that I had a way to specify priorities as I was inserting values into a treap.

### CODE REVIEW

File: pfds/scribblings/treap.scrbl [hunk 1]
```diff
@@ -26,7 +26,7 @@ operations @racket[insert], @racket[find-min/max] and
 
 @defproc[(treap [comp (A A -> Boolean)] [a A] ...) (Treap A)]{
 Function @racket[treap] creates a Treap with the given 
-inputs.    
+inputs. Elements are assigned a random between from 0 to 1.
```
Comment (takikawa) [inline pfds/scribblings/treap.scrbl]: Missing word "priority" here?

File: pfds/scribblings/treap.scrbl [hunk 2]
```diff
@@ -215,4 +226,68 @@ function takes an extra comparison function.
...
+appropriate priorities if one wants to avoid @bold{@italic{O(n)}} running times.
+
+@defproc[(treap/priority [comp (A A -> Boolean)] [a (Pairof A Real)] ...) (Treap A)]{
+Function @racket[treap] creates a Treap with the given inputs each assigned the priority
+they are paired with.
+@examples[#:eval evaluate
+(treap/priority < '(1 . 3) '(2 . 2) '(3 . 8) '(4 . 7) '(5 . 1) '(6 . 4))
+]
+
+In the above example, the treap obtained will have elements 1 thru' 6
```
Comment (takikawa) [inline pfds/scribblings/treap.scrbl]: Minor: I think the `'` is a typo?

**Comment by takikawa** [comment]
Thanks! Could you also add some unit tests for the new operations in the treap unit test file? Otherwise looks good and I can merge it.

**Comment by dented42** [comment]
I believe that's everything.

**Comment by takikawa** [comment]
Thanks!

**Pull request merged**